src/Lib/Managers/NobManager.vala

```diff
@@ -0,0 +1,345 @@
+        this.selected_nob = selected_nob;
+    }
+
+    public Nob get_selected_nob () {
+        return selected_nob;
+    }
+
+    public Nob get_grabbed_id (Goo.CanvasItem? target) {
+        int grabbed_id = -1;
+
+        for (int i = 0; i < 9; i++) {
+            if (target == nobs[i]) grabbed_id = i;
+        }
+
+        return (Nob) grabbed_id;
+    }
+
+    private void on_zoom (double _current_scale) {
+        current_scale = _current_scale;
+    }
+
+    private void update_select_bb_coords (List<Models.CanvasItem> selected_items) {
+        // Bounding box edges
+        double bb_left = 1e6, bb_top = 1e6, bb_right = 0, bb_bottom = 0;
+
+        foreach (var item in selected_items) {
+            Goo.CanvasBounds item_bounds;
+            item.get_bounds (out item_bounds);
+
+            bb_left = double.min (bb_left, item_bounds.x1);
+            bb_top = double.min (bb_top, item_bounds.y1);
+            bb_right = double.max (bb_right, item_bounds.x2);
+            bb_bottom = double.max (bb_bottom, item_bounds.y2);
+        }
+
+        select_bb = Goo.CanvasBounds () {
+            x1 = bb_left,
+            y1 = bb_top,
+            x2 = bb_right,
+            y2 = bb_bottom
+        };
+
+        top = select_bb.y1;
+        left = select_bb.x1;
+        width = select_bb.x2 - select_bb.x1;
+        height = select_bb.y2 - select_bb.y1;
+    }
+
+    private void on_add_select_effect (List<Models.CanvasItem> selected_items) {
+        remove_select_effect ();
+
+        if (selected_items.length () == 0) {
+            return;
+        }
+
+        if (selected_items.length () > 1) {
+            update_select_bb_coords (selected_items);
+        }
+
+        update_select_effect (selected_items);
+        update_nob_position (selected_items);
+    }
+
+    private void remove_select_effect (bool keep_selection = false) {
+        if (select_effect == null) {
+            return;
+        }
+
+        select_effect.remove ();
+        select_effect = null;
+
+        for (int i = 0; i < 9; i++) {
+            nobs[i].remove ();
+        }
+    }
+
+    private void update_select_effect (List<Models.CanvasItem> selected_items) {
+        double x = 0.0;
+        double y = 0.0;
+        double line_width = 0.0;
+        double width = 0.0;
+        double height = 0.0;
+
+        var transform = Cairo.Matrix.identity ();
+
+        set_bound_coordinates (
+            selected_items,
+            ref x, ref y,
+            ref transform,
+            ref line_width,
+            ref width, ref height
+        );
+
+        // Account for line_width
+        x -= line_width;
+        y -= line_width;
+        width += line_width * 2;
+        height += line_width * 2;
+
+        select_effect = new Goo.CanvasRect (
+            null,
+            x, y,
+            width,
+            height,
+            "line-width", 1.0 / current_scale,
+            "stroke-color", "#666",
+            null
```

Comment:
Set the line width and stroke color variables as constants at the beginning of the class.